snuba/web/query.py

```diff
@@ -209,33 +268,69 @@ def log_query_and_update_stats(
     """
 
     stats.update(query_settings)
+
+    query_metadata.query_list.append(
+        ClickhouseQueryMetadata(
+            sql=sql, timer=timer, stats=stats, status=status, trace_id=trace_id,
```

Comment:
This is the same timer of the main query, which also means all subqueries will have the same timer. Is this intended ?What do you plan to record ?

Reply:
Yeah, I think it could make sense for each clickhouse query to have it's own timer, but I think that would require a much larger refactor.

Right now, the timer works in a weird way, this records the timings each time `update_query_metadata_and_stats` gets called. The duration on each split query is always being calculated from the first mark though. So if there are 2 queries that take 1 second each we'll have a query list that looks like:
```
clickhouse_queries = [{{"timing": {"duration": 1000}}, {"timing": {"duration":  2000}}]
```
(yes i know this doesn't exactly make sense, but I don't want to address it in this PR, it is the same way that metrics works right now)